### ISSUE TITLE
Failing Test: SaveParticipantTest.test3

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform#651]
org.eclipse.core.tests.resources.session.SaveParticipantTest

the test is not used for  almost 20 years. It was once listed as comment in org.eclipse.core.tests.resources.session.AllSessionTests 
from [2d6f69d722d1fb50ad084fc05d3d08c0e3ceac7c](https://github.com/eclipse-platform/eclipse.platform/commit/2d6f69d722d1fb50ad084fc05d3d08c0e3ceac7c#diff-e8f0c12f0f1fdf6b16f0e1c6b5f4c055c1cf672628a4de81838b31e5f5541261R45)

until [31a4b42f024e545077240af22e1349e34425ed07](https://github.com/eclipse-platform/eclipse.platform/commit/31a4b42f024e545077240af22e1349e34425ed07#diff-e8f0c12f0f1fdf6b16f0e1c6b5f4c055c1cf672628a4de81838b31e5f5541261L53)

**Comment by jukzi** [comment]
```
junit.framework.AssertionFailedError: 6.0
	at junit.framework.Assert.fail(Assert.java:57)
	at junit.framework.TestCase.fail(TestCase.java:223)
	at org.eclipse.core.tests.resources.saveparticipant.SaveManager3Test.testSaveParticipant(SaveManager3Test.java:143)
	at org.eclipse.core.tests.resources.saveparticipant.SaveManagerTest.test3(SaveManagerTest.java:104)
```

logging
```
delta is different than expected
Verifying delta for /
Verifying delta for /MyProject
Verifying delta for /MyProject/addedFile
Verifying delta for /Project2
Verifying delta for /Project2/.project
Verifying delta for /Project2/.settings
	Missing expected change
		Actual kind: <ADDED>
		Actual change flags: <0>
Verifying delta for /Project2/.settings/org.eclipse.core.resources.prefs
	Missing expected change
		Actual kind: <ADDED>
		Actual change flags: <0>
Verifying delta for /Project2/addedFile
Verifying delta for /Project2/file110
Verifying delta for /Project2/folder110
Verifying delta for /Project2/folder110/file120
Verifying delta for /Project2/folder111
Verifying delta for /Project2/folder111/file121

```

**Comment by jukzi** [comment]
regression from https://bugs.eclipse.org/bugs/show_bug.cgi?id=479450

**Comment by HeikoKlare** [comment]
fixed via #652
